internal/resource/state.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package resource
```

Comment:
todo: This belongs in the core/resource package instead of internal. It's a entity without dependencies on other entities.